File: web/web-frontend/src/app/manual-tests/runner/model/operation/update-manual-test-execution.model.ts
import {Path} from "../../../../model/infrastructure/path/path.model";
import {JsonUtil} from "../../../../utils/json.util";
import {ManualTestModel} from "../../../model/manual-test.model";
import {ManualTestExeModel} from "../manual-test-exe.model";

export class UpdateManualTestExecutionModel implements Serializable<UpdateManualTestExecutionModel> {

    manualTestRunnerPath: Path;
    manualTestExe: ManualTestModel;

    constructor(manualTestRunnerPath: Path, manualTestExeModel: ManualTestExeModel) {
        this.manualTestRunnerPath = manualTestRunnerPath;
        this.manualTestExe = manualTestExeModel;
    }

    deserialize(input: Object): UpdateManualTestExecutionModel {
        this.manualTestRunnerPath = Path.deserialize(input["manualTestRunnerPath"]);
        this.manualTestExe = input['manualTestExe'];

        return this;
    }

    serialize() {
        return "" +
            '{' +
            '"manualTestRunnerPath":' + JsonUtil.serializeSerializable(this.manualTestRunnerPath) + ',' +
            '"manualTestExe":' + JsonUtil.serializeSerializable(this.manualTestExe) +
            '}'
    }
}

File: model/src/main/java/com/testerum/model/manual/runner/operation/UpdateManualTestExecutionModel.kt
package com.testerum.model.manual.runner.operation

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.model.infrastructure.path.Path
import com.testerum.model.manual.runner.ManualTestExe

data class UpdateManualTestExecutionModel @JsonCreator constructor(
        @JsonProperty("manualTestRunnerPath") val manualTestRunnerPath: Path,
        @JsonProperty("manualTestExe") val manualTest: ManualTestExe)